website/models.py

```diff
@@ -1,129 +1,132 @@
-
-
-#  TODO institutions model
+from django.db import models
+from django.contrib.auth.models import User
+
+
+class Doctor(User):
+    """Let's assume a username is a doctor's number needed for receipt."""
+    TITLES = (
+        ('LE', 'lek.'),
+        ('DR', 'dr n. med.'),
+        ('DH', 'dr hab. n. med.'),
+        ('PR', 'prof. dr hab. n. med.'),
+    )
+    gender = models.BooleanField()  # true for Mr./false for Ms.
+    title = models.CharField(max_length=2, choices=TITLES, default=TITLES[0][0])
+
+    def __str__(self):
+        return self.title + ' ' + self.first_name + ' ' + self.last_name
+
+
+class Patient(User):
+    """Let's assume a username is a pesel number."""
+    gender = models.BooleanField() # true for man
+    pesel = models.PositiveIntegerField(primary_key=True)
+    date_of_birth = models.DateField()
+    conditions = models.ManyToManyField('Condition')
+
+    #I put in comments the whole class meta because I couldn't be bothered to deal with errors.
+    """class Meta:
+        indexes = [
+            models.Index(fields=['first_name', 'last_name'])
+        ]"""
+    def __str__(self):
+        return self.first_name + ' ' + self.last_name
+
+
+class Condition(models.Model):
+    """A condition of a patient making him eligible for a refund."""
+    id = models.PositiveIntegerField(primary_key=True)
+    name = models.CharField(max_length=150)
+
+    def __str__(self):
+        return self.name
+
+
+class StoryEntry(models.Model):
+    """Single entry in a patient's story. Holds information either only about a disease
+       or about a disease and a prescription."""
+    patient = models.ForeignKey('Patient', on_delete=models.CASCADE)
+    doctor = models.ForeignKey('Doctor', on_delete=models.CASCADE)#Can not be null on delete.
+    date = models.DateTimeField(auto_now_add=True)
+    #story = models.ForeignKey('Story', on_delete=models.CASCADE) no 'Story' model. Is it necessary?
+    disease = models.ForeignKey('Disease', on_delete=models.CASCADE)#Can not be null on delete.
+    """I changed a name for prescription because it clashed with the name of Prescription.StoryEntry (line 70)
+    feel free to change it"""
+    prescription1 = models.ForeignKey('Prescription', null=True, on_delete=models.SET_NULL)
+
+
+class Disease(models.Model):
+    """Disease coded with ICD 10"""
+    code = models.CharField(max_length=5, primary_key=True)
+    name = models.CharField(max_length=150)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=['name'])
+        ]  
+    def __str__(self):
+        return self.name
+
+
+class Prescription(models.Model):
+    id = models.PositiveIntegerField(primary_key=True)
+    story = models.ForeignKey('StoryEntry', on_delete=models.CASCADE)
+
+
+class PrescriptionEntry(models.Model):
+    prescription = models.ForeignKey('Prescription', on_delete=models.CASCADE)
+    #Neither medication nor dosage can be null on delete, I changed it to cascade for now.

```

Comment:
Medication niech zostanie on delete cascade, bo jeśli usunie się lek to recepta nie ma sensu, ale lek bez dawki już i tak jest cenną informacją dla lekarza.